x/gamm/pool-models/stableswap/amm_test.go

```diff
@@ -92,93 +340,37 @@ func TestCFMMInvariantTwoAssets(t *testing.T) {
 func TestCFMMInvariantMultiAssets(t *testing.T) {
 	kErrTolerance := osmomath.OneDec()
 
-	tests := map[string]struct {
-		xReserve    osmomath.BigDec
-		yReserve    osmomath.BigDec
-		uReserve    osmomath.BigDec
-		wSumSquares osmomath.BigDec
-		yIn         osmomath.BigDec
-		expectPanic bool
-	}{
-		"4-asset pool, small input": {
-			osmomath.NewBigDec(100),
-			osmomath.NewBigDec(100),
-			// represents a 4-asset pool with 100 in each reserve
-			osmomath.NewBigDec(200),
-			osmomath.NewBigDec(20000),
-			osmomath.NewBigDec(1),
-			false,
-		},
-		"4-asset pool, large input": {
-			osmomath.NewBigDec(100),
-			osmomath.NewBigDec(100),
-			osmomath.NewBigDec(200),
-			osmomath.NewBigDec(20000),
-			osmomath.NewBigDec(1000),
-			false,
-		},
-		// This test fails due to a bug in our original solver
-		// "large pool, large input": {
-		// 	sdk.NewDec(100000),
-		// 	sdk.NewDec(100000),
-		// 	sdk.NewDec(10000),
-		// },
+	// TODO: switch solveCfmmMulti to binary search and replace this with test case suite
```

Comment:
Sorry not sure I'm following the plan here. 

If we are going to replace it with the binary search why would we have two sets of tests?

Reply:
Originally we had 3 distinct solvers for our CFMM (general direct, simplified direct, and binary search), so we kept `solveCfmm` and `solveCfmmMulti` as a layer of abstraction so we wouldn't have to manually edit all pool logic as we worked on the different solver types

If/when we decide to enshrine one solver as our main one (likely binary search as indicated in here), then it might make sense to consolidate these into one, but I think we should do that after all of the loose ends are tied with our solvers

Reply:
Please see this issue: https://github.com/osmosis-labs/osmosis/issues/2845